### PR TITLE
refator schematics workspace api

### PR DESCRIPTION
**Pull request by damyanpetev**
Closes # .  

Additional information related to this pull request:

### CODE REVIEW

File: packages/cli/migrations/update-5_0_3/index.ts [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
 
 export default function(): Rule {
-	return (host: Tree, context: SchematicContext) => {
-		const workspace = getWorkspace(host);
-		const project = workspace.defaultProject ?
-			workspace.projects[workspace.defaultProject] :
-			workspace.projects[0];
+	return async (tree: Tree, context: SchematicContext) => {
+		const { workspace } = await workspaces.readWorkspace("/", createWorkspaceHost(tree));
+		const project = workspace.extensions.defaultProject ?
```
Comment (jackofdiamond5) [inline packages/cli/migrations/update-5_0_3/index.ts:9]: You caн use `getDefaultProject` here.

File: packages/cli/migrations/update-4_2_3/index.ts [hunk 2]
```diff
@@ -16,13 +16,13 @@ const paginatorStyleReplacement =
...
 
 export default function(): Rule {
-	return (host: Tree, context: SchematicContext) => {
-		const workspace = getWorkspace(host);
-		const project = workspace.defaultProject ?
-			workspace.projects[workspace.defaultProject] :
-			workspace.projects[0];
+	return async (tree: Tree, context: SchematicContext) => {
+		const { workspace } = await workspaces.readWorkspace("/", createWorkspaceHost(tree));
+		const project = workspace.extensions.defaultProject ?
```
Comment (jackofdiamond5) [inline packages/cli/migrations/update-4_2_3/index.ts:21]: You can use `getDefaultProject` here.

File: packages/cli/migrations/update-3_2/index.ts [hunk 3]
```diff
@@ -14,13 +14,13 @@ const summaryLatestReplacement =
...
 
 export default function(): Rule {
-	return (host: Tree, context: SchematicContext) => {
-		const workspace = getWorkspace(host);
-		const project = workspace.defaultProject ?
-			workspace.projects[workspace.defaultProject] :
-			workspace.projects[0];
+	return async (tree: Tree, context: SchematicContext) => {
+		const { workspace } = await workspaces.readWorkspace("/", createWorkspaceHost(tree));
+		const project = workspace.extensions.defaultProject ?
```
Comment (jackofdiamond5) [inline packages/cli/migrations/update-3_2/index.ts:19]: You can use `getDefaultProject` here.

File: packages/cli/migrations/update-5_0_0/index.ts [hunk 4]
```diff
@@ -36,18 +35,18 @@ export function setVirtual(tree: Tree) {
...
 
 export default function(): Rule {
-	return (host: Tree, context: SchematicContext) => {
-		const workspace = getWorkspace(host);
-		const project = workspace.defaultProject ?
-			workspace.projects[workspace.defaultProject] :
-			workspace.projects[0];
+	return async (tree: Tree, context: SchematicContext) => {
+		const { workspace } = await workspaces.readWorkspace("/", createWorkspaceHost(tree));
+		const project = workspace.extensions.defaultProject ?
```
Comment (jackofdiamond5) [inline packages/cli/migrations/update-5_0_0/index.ts:40]: You can use `getDefaultProject`  here.

File: packages/ng-schematics/src/utils/theme-import.ts [hunk 5]
```diff
@@ -53,78 +53,52 @@ function importDefaultThemeSass(tree: Tree, ext: string): Tree {
...
-export function getDefaultProject(tree: Tree): WorkspaceProject<ProjectType> {
-	const workspace = getWorkspace(tree);
-	return workspace.projects[workspace.defaultProject];
+	// workspace tracks changes internally:
+	await workspaces.writeWorkspace(workspace, host);
 }
 
-export function getDefaultProjectBuildOptions(tree: Tree) {
-	return getTargetedProjectOptions(getDefaultProject(tree), "build");
+export async function getDefaultProject(tree: Tree): Promise<workspaces.ProjectDefinition> {
```
Comment (jackofdiamond5) [inline packages/ng-schematics/src/utils/theme-import.ts:77]: How about we mode this one to a util file where we can access it anywhere we need it. `core/util/Schematics.ts` could be a good place for the time being, no? 

We could also create a function that returns not only the default project but all projects in a workspace, with the default project being one of them, this should prove a tiny bit more robust.

File: packages/ng-schematics/src/utils/theme-import.ts [hunk 6]
```diff
@@ -53,78 +53,52 @@ function importDefaultThemeSass(tree: Tree, ext: string): Tree {
...
-			workspace.projects[projectName].architect[key].options = {};
-		}
-		if (!workspace.projects[projectName].architect[key].options.styles) {
-			workspace.projects[projectName].architect[key].options.styles = [];
-		}
-		if (!workspace.projects[projectName].architect[key].options.styles.includes(cssImport)) {
-			workspace.projects[projectName].architect[key].options.styles.push(cssImport);
-			return true;
-		}
+	const project = workspace.extensions.defaultProject ?
```
Comment (jackofdiamond5) [inline packages/ng-schematics/src/utils/theme-import.ts:88]: You can use `getDefaultProject` here.

File: packages/ng-schematics/src/utils/theme-import.ts [hunk 7]
```diff
@@ -53,78 +53,52 @@ function importDefaultThemeSass(tree: Tree, ext: string): Tree {
...
+		workspace.projects.get(workspace.extensions.defaultProject as string) :
+		workspace.projects.values().next().value as workspaces.ProjectDefinition;
+
+	const target = project?.targets.get(key);
+	if (!target) {
+		// TODO: Log target not found
```
Comment (jackofdiamond5) [inline packages/ng-schematics/src/utils/theme-import.ts:94]: Should we log it? Can `project.targets` be `undefined`?

**Review by Lipata — APPROVED** [review]

**Review by Lipata — APPROVED** [review]

**Pull request merged**